### ISSUE TITLE
IPI fails on iRMC servers due to the loss of Ignition file

### ISSUE THREAD

**Issue by hs0210** [openshift/installer#5542]
# Version

```console
openshift-baremetal-install version
openshift-baremetal-install 4.10.0-0.nightly-2022-01-13-061145
built from commit 130a5673f2a431ca96e603d55be7c8865a1b0849
release image quay.io/openshift-release-dev/ocp-release-nightly@sha256:c03f58e1f3000ddeabcfeb61f343fba640e9354e2e5347c044b41edb4faf7b96
release architecture amd64
```

# Platform:

IPI

# What happened?

Inspect failed. And from ServerView, we saw that the Ignition File is missing.
![Screenshot from 2022-01-17 17-36-30](https://user-images.githubusercontent.com/53652350/149751267-09b1b1a7-ca48-42d6-b97f-875679ad1436.png)
![Screenshot from 2022-01-17 17-36-52](https://user-images.githubusercontent.com/53652350/149751294-868cbfac-5427-4f28-ab9e-6458c878edeb.png)

# What you expected to happen?

IPI can successfully run on iRMC.

**Comment by zhouhao3** [comment]
The reason there is no Ignition file should be that [configure-coreos-ipa](https://github.com/openshift/ironic-image/blob/master/scripts/configure-coreos-ipa) is not called in [httpd](https://github.com/openshift/ironic-image/blob/master/scripts/runhttpd) and the [IGNITION_FILE](https://github.com/openshift/ironic-image/blob/master/scripts/coreos-ipa-common.sh#L4) is not created, So the address of IGNITION_FILE is not [added](https://github.com/openshift/ironic-image/blob/master/scripts/coreos-ipa-common.sh#L9-L11) to the boot kernel parameters.

**Comment by zaneb** [comment]
> the address of IGNITION_FILE is not [added](https://github.com/openshift/ironic-image/blob/master/scripts/coreos-ipa-common.sh#L9-L11) to the boot kernel parameters.

It's not supposed to be. The ignition is bundled into the initramfs file by the image-customization-controller, not passed as a URL on the kernel command line. In fact, it *can't* be passed as a URL because the Ignition potentially contains the network configuration needed to reach any URLs.

**Comment by zaneb** [comment]
I wonder if this can be explained by a wrong/missing MAC address: https://github.com/openshift/installer/pull/5541#issuecomment-1015649857

**Comment by hs0210** [comment]
> I wonder if this can be explained by a wrong/missing MAC address: #5541 (comment)

The MAC address is correct. It seems that the reason is ironic didn't create the pxe config correctly. see details at [zhou's comment in #5541](https://github.com/openshift/installer/pull/5541#issuecomment-1016120434)

**Comment by zaneb** [comment]
Is this resolved by changing the boot interface as in https://github.com/openshift/installer/pull/5541#issuecomment-1017155768?

**Comment by hs0210** [comment]
> Is this resolved by changing the boot interface as in #5541 (comment)?

Thanks, it works, I'll close this issue.
